tamr_toolbox/dataset/manage.py

```diff
@@ -0,0 +1,527 @@
+from typing import List, Optional, Dict, Iterable
+import logging
+
+from tamr_unify_client import Client
+from tamr_unify_client.dataset.resource import Dataset
+from tamr_toolbox.models import attribute_type
+
+from tamr_toolbox.models.data_type import JsonDict
+
+LOGGER = logging.getLogger(__name__)
+
+
+def exists(*, client: Client, dataset_name: str) -> bool:
+    """
+    Check if a dataset exists in a Tamr instance
+
+    Args:
+        client: Tamr python client object for the target instance
+        dataset_name: The dataset name
+
+    Return:
+        True or False for if the dataset exists in target instance
+    """
+
+    try:
+        client.datasets.by_name(dataset_name)
+    except KeyError:
+        return False
+
+    return True
+
+
+def create(
+    *,
+    client: Client,
+    dataset_name: str,
+    dataset: Optional[Dataset] = None,
+    primary_keys: Optional[List[str]] = None,
+    attributes: Optional[Iterable[str]] = None,
+    attribute_types: Optional[Dict[str, attribute_type.AttributeType]] = None,
+    attribute_descriptions: Optional[Dict[str, str]] = None,
+    description: Optional[str] = None,
+    external_id: Optional[str] = None,
+    tags: Optional[List[str]] = None,
+) -> Dataset:
+    """
+    Flexibly create a source dataset in Tamr. A template dataset object can be passed in to create
+    a duplicate dataset with a new name. If the template dataset is not provided, the user must
+    define the primary_keys for the dataset to be created. Additional attributes can be added
+    in the attributes argument. The default attribute type will be ARRAY STRING. To set non-default
+    attribute types, you must define them in the attribute_types dictionary. Any attribute
+    descriptions can be specified in the attribute_descriptions dictionary.
```

Comment:
nit: wording and consistent grammar
```suggestion
    """Flexibly create a source dataset in Tamr

    A template dataset object can be passed in to create a duplicate dataset with a new name. If
    the template dataset is not provided, the primary_keys must be defined for the dataset to be
    created. Additional attributes can be added in the attributes argument. The default attribute
    type will be ARRAY STRING. Non-default attribute types can be specified in the attribute_types
    dictionary. Any attribute descriptions can be specified in the attribute_descriptions 
    dictionary.
```